pepper-apis/src/main/java/org/broadinstitute/ddp/route/JoinMailingListRoute.java

```diff
@@ -50,6 +56,34 @@ public Object handle(Request request, Response response, JoinMailingListPayload
             JdbiMailingList jdbiMailingList = handle.attach(JdbiMailingList.class);
 
             String info = payload.getInfo() != null && payload.getInfo().size() > 0 ? String.join(", ", payload.getInfo()) : null;
+            String languageCode = payload.getLanguageCode();
+            Long languageCodeId = null;
+            if (StringUtils.isEmpty(languageCode)) {
+                LanguageDto preferredUserLangDto = RouteUtil.getUserLanguage(request);
+                if (preferredUserLangDto != null) {
+                    languageCode = preferredUserLangDto.getIsoCode();
+                }
+            }
+
+            if (StringUtils.isNotEmpty(languageCode)) {
+                languageCodeId = LanguageStore.get(languageCode).getId();
```

Comment:
Might want to wrap this in null-check in case client sends request with an unknown language code, in which case I suggest we return a `400`.